src/OrchardCore.Build/Dependencies.props

```diff
@@ -42,6 +42,7 @@
     <PackageManagement Include="Microsoft.Extensions.Azure" Version="1.7.3" />
     <PackageManagement Include="Microsoft.Extensions.Http.Resilience" Version="8.4.0" />
     <PackageManagement Include="Microsoft.Identity.Web" Version="2.18.1" />
+    <PackageManagement Include="Microsoft.IdentityModel.Protocols.OpenIdConnect" Version="7.5.1" />
```

Comment:
Add a comment here on why this is necessary, also linking to this PR with the rest of the explanation.

We should be able to remove it once the Microsoft package updates its dependency, right? Then also note that so we'll be able to keep it it in mind.

Reply:
> We should be able to remove it once the Microsoft package updates its dependency, right? Then also note that so we'll be able to keep it it in mind.

No, the ASP.NET team never updates the IM package version in patch versions (and minor versions are no longer a thing in .NET or ASP.NET Core). It's sadly here to stay.

Reply:
I see , but eventually, they'll update it, no? Unless 8.0.4 is the last version of `Microsoft.AspNetCore.Authentication.OpenIdConnect ` ever.

Reply:
No, `Microsoft.AspNetCore.Authentication.OpenIdConnect` 8.0.5 just shipped today (with .NET 8.0.5) and it's still using the same (old) IdentityModel version: https://www.nuget.org/packages/Microsoft.AspNetCore.Authentication.OpenIdConnect/8.0.5#dependencies-body-tab

The ASP.NET team doesn't update dependencies like IM in patch versions. Unless they agree to change this policy, you'll have to wait .NET 9.0 for that dependency to be bumped.

Reply:
So they'll eventually update it :). I'm not talking about just patch versions, but any future version. So, eventually, even if that's just after .NET 9, we'll need to remove this reference (not just for tidiness but because at that point it'll be most possibly outdated). Having a comment about it here would help us remember this (with the keyword "net8.0" so when searching for it during the next .NET upgrade we'll notice it).

I just now realized that there's no entry for `Microsoft.AspNetCore.Authentication.OpenIdConnect` in `Dependencies.props`, which is strange though. This second comment would be more appropriate on it, since that'll be updated and thus will the line change.

Reply:
> So they'll eventually update it :). I'm not talking about just patch versions, but any future version. So, eventually, even if that's just after .NET 9, we'll need to remove this reference (not just for tidiness but because at that point it'll be most possibly outdated). Having a comment about it here would help us remember this (with the keyword "net8.0" so when searching for it during the next .NET upgrade we'll notice it).

IM ships once or twice a month: so sure, .NET 9.0 will probably be released with an up-to-date IM dependency, but it will quickly go out of sync like .NET 8.0 and all the previous .NET releases before it (again, unless the ASP.NET team decides to change their policy, as requested by the IM team). There would be no point removing the reference just to bring it back one or two months after 🤣 

> I just now realized that there's no entry for `Microsoft.AspNetCore.Authentication.OpenIdConnect` in `Dependencies.props`, which is strange though. This second comment would be more appropriate on it, since that'll be updated and thus will the line change.

That one is defined in `Dependencies.AspNetCore.props`.

Reply:
I really don't understand what we're debating :). `Microsoft.AspNetCore.Authentication.OpenIdConnect` references an older version of `Microsoft.IdentityModel.Protocols.OpenIdConnect` currently, so we need this fix. In the future,  `Microsoft.AspNetCore.Authentication.OpenIdConnect` will reference a newer version of `Microsoft.IdentityModel.Protocols.OpenIdConnect`. At that point we'll need to remove this manual reference, or, we'll need to keep this reference up-to-date with the version that `OpenIddict.Server.AspNetCore` uses.

Thus, this is not a usual package reference, because it's a workaround that you have to understand the reason for, and keep this in mind when updating versions.

All I want is that if in a year from now somebody submits a PR [like this one](https://github.com/OrchardCMS/OrchardCore/pull/16063) with just the version numbers changed, we'll know what to look out for. So, please add a corresponding comment in `Dependencies.AspNetCore.props` as well. Otherwise, somebody just updating one package will have tunnel vision and miss the bigger picture.

Reply:
> At that point we'll need to remove this manual reference, or, we'll need to keep this reference up-to-date with the version that `OpenIddict.Server.AspNetCore` uses.

As I said, only the second option is realistic: even if `Microsoft.AspNetCore.Authentication.OpenIdConnect` 9.0 ships with a recent IM dependency, it will be frozen and all the subsequent versions (e.g 9.0.1, 9.0.2, etc.) will keep using it instead of moving to the latest available.

> All I want is that if in a year from now somebody submits a PR [like this one](https://github.com/OrchardCMS/OrchardCore/pull/16063) with just the version numbers changed, we'll know what to look out for. So, please add a corresponding comment in `Dependencies.AspNetCore.props` as well. Otherwise, somebody just updating one package will have tunnel vision and miss the bigger picture.

I can certainly do that if you want me to, but it's unnecessary: let's say you update `Microsoft.AspNetCore.Authentication.OpenIdConnect` to an imaginary version referencing an IM version newer than the one used for OpenIddict: because the OpenID module explicitly references `Microsoft.IdentityModel.Protocols.OpenIdConnect`, you'd immediately get a NuGet warning telling you have a conflict you need to fix.

Reply:
> Thus, this is not a usual package reference, because it's a workaround that you have to understand the reason for, and keep this in mind when updating versions.

BTW, it's not really a "workaround" but just how NuGet and its "direct dependency always wins" policy work.

More info here if you're interested:
  - https://learn.microsoft.com/en-us/nuget/concepts/dependency-resolution#direct-dependency-wins
  - https://learn.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu1605

Reply:
Which NuGet warning are you referring to? To my knowledge, you need something like [`dotnet consolidate`](https://github.com/olsh/dotnet-consolidate) to pinpoint such issues but a normal .NET build won't emit any warnings.

Reply:
The one I mentioned in my last message: NU1605.

Here's an example referencing both `OpenIddict.Server.AspNetCore` (that uses IM 7.5.1) and IM 7.5.0:

![image](https://github.com/OrchardCMS/OrchardCore/assets/6998306/fef11d45-4dd3-437e-9e57-72f421938730)

Regarding the third-party `dotnet consolidate` tool, it's not really useful for projects that use centralized package versions management (either the built-in NuGet feature, used by OpenIddict) or the equivalent we have here in Orchard.

Reply:
OK, great then, thanks for explaining!

Reply:
My pleasure 😄